### PR TITLE
refactor: `livechat rooms` - 1

### PR DESCRIPTION
**Pull request by KevLehman**
ARCH-928

## Further comments

Basically changes findOneById and that's being used in a ton of files

### CODE REVIEW

File: apps/meteor/app/livechat/server/hooks/sendToCRM.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-import { Messages } from '@rocket.chat/models';
+import { LivechatRooms, Messages } from '@rocket.chat/models';
 
 import { settings } from '../../../settings/server';
 import { callbacks } from '../../../../lib/callbacks';
-import { LivechatRooms } from '../../../models/server';
 import { Livechat } from '../lib/Livechat';
 import { normalizeMessageFileUpload } from '../../../utils/server/functions/normalizeMessageFileUpload';
 
+// TODO: remove Promise.await from this file when callbacks are async
```
Comment (sampaiodiego) [inline apps/meteor/app/livechat/server/hooks/sendToCRM.js:9]: callbacks are async already =)

File: apps/meteor/app/livechat/server/lib/LivechatTyped.ts [hunk 2]
```diff
@@ -120,6 +120,7 @@ class LivechatClass {
 			transcriptRequested: !!transcriptRequest,
 		};
 
+		// TODO: properly validate that room exists before using instead of typecast :)
```
Comment (sampaiodiego) [inline apps/meteor/app/livechat/server/lib/LivechatTyped.ts:123]: are you going to do this now? 🙏

File: apps/meteor/app/livechat/server/startup.ts [hunk 3]
```diff
@@ -18,7 +18,8 @@ import { RoutingManager } from './lib/RoutingManager';
 import './roomAccessValidator.internalService';
 
 Meteor.startup(async () => {
-	roomCoordinator.setRoomFind('l', (_id) => LivechatRooms.findOneById(_id));
+	// TODO: remove promise.await when roomCoordinator supports async functions
```
Comment (sampaiodiego) [inline apps/meteor/app/livechat/server/startup.ts:21]: roomCoordinator supports async now =)

File: apps/meteor/ee/app/livechat-enterprise/server/hooks/resumeOnHold.js [hunk 4]
```diff
@@ -26,8 +26,9 @@ const handleAfterSaveMessage = (message, roomParams) => {
 		return message;
 	}
 
+	// TODO: remove promise.await
```
Comment (sampaiodiego) [inline apps/meteor/ee/app/livechat-enterprise/server/hooks/resumeOnHold.js:29]: you can remove it now

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: apps/meteor/app/livechat/server/hooks/sendToCRM.js (hunk 1 above)
Comment (KevLehman) [reply apps/meteor/app/livechat/server/hooks/sendToCRM.js:9]: image

File: apps/meteor/app/livechat/server/startup.ts (hunk 3 above)
Comment (KevLehman) [reply apps/meteor/app/livechat/server/startup.ts:21]: U sure? The file doesn't have any async thing in it yet 🤔

File: apps/meteor/app/livechat/server/lib/LivechatTyped.ts (hunk 2 above)
Comment (KevLehman) [reply apps/meteor/app/livechat/server/lib/LivechatTyped.ts]: I mean, why not

File: apps/meteor/app/livechat/server/startup.ts [hunk 5]
```diff
@@ -18,6 +18,7 @@ import { RoutingManager } from './lib/RoutingManager';
 import './roomAccessValidator.internalService';
 
 Meteor.startup(async () => {
+	// TODO: remove promise.await when roomCoordinator supports async functions
```
Comment (sampaiodiego) [inline apps/meteor/app/livechat/server/startup.ts:21]: I guess this comment is not required anymore, right?

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**